### PR TITLE
Styling fixes for ReportPreview/MoneyRequestPreview

### PR DESCRIPTION
**Pull request by grgia**
### Details

**ReportPreview**
✅ Fixes Text line height
✅ Fixes Gaps
✅ Fixes Counter Highlight Color
imageimage

**MoneyRequestPreview**
✅ Fixes Text line height
✅ Fixes Gaps
image

**ReportPreview Subtitle Copy Change**
✅ Add support for displaying `# of pending` in the Report Preview
image

### Fixed Issues

$ https://github.com/Expensify/App/issues/33979
PROPOSAL:

### Tests

- [X] Verify that no errors appear in the JS console
- [X] See QA steps 
Note for C+: to add a pending card transaction, simply set transaction.status to 'Pending' for a transaction on the report via onyx console 

- [X] QA'd UI with design

### QA Steps

- [X] Verify that no errors appear in the JS console

- Create a report with 2 manual requests, go to the parent chat report and locate the report preview
- [X] Verify that copy reads "2 requests"
- Add 1 Pending ECard transaction
- [X] Verify that copy reads "2 requests, 1 pending"
- Add 1 Scanning transaction
- [X] Verify that copy reads "2 requests, 1 scanning, 1 pending"
- Delete the ECard transaction
- [X] Verify that copy reads "2 requests, 1 scanning"

### PR Author Checklist

- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [X] I added steps for local testing in the `Tests` section
    - [X] I added steps for the expected offline behavior in the `Offline steps` section
    - [X] I added steps for Staging and/or Production testing in the `QA steps` section
    - [X] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I ran the tests on **all platforms** & verified they passed on:
    - [X] Android: Native
    - [X] Android: mWeb Chrome
    - [X] iOS: Native
    - [X] iOS: mWeb Safari
    - [X] MacOS: Chrome / Safari
    - [X] MacOS: Desktop
- [X] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [X] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] I verified that if a function's arguments changed that all usages have also been updated correctly
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [X] If the PR modifies the form input styles:
    - [X] I verified that all the inputs inside a form are aligned with each other.
    - [X] I added `Design` label so the design team can review the changes.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

### CODE REVIEW

**Comment by grgia** [comment]
@shawnborton would you mind taking a look at the screenshots I added? Let me know if anything else needs changing!

**Comment by shawnborton** [comment]
Looking good! The only thing I see is that the green button feels just slightly far away from the text above it? How much space is there? 

cc @Expensify/design for a quick review too!

**Comment by grgia** [comment]
@shawnborton there's a gap of 16 and it looks like the button has a margin-top of 12px

image

**Comment by shawnborton** [comment]
Okay cool, I think we just want the 16px gap between text and button, so we can get rid of the extra 12px of padding above the button. This is what I'm seeing in Figma at least:
CleanShot 2024-02-12 at 08 42 40@2x

**Comment by grgia** [comment]
@shawnborton @dannymcclain updated!

**Comment by shawnborton** [comment]
Feeling good to me! I just fired up a test build so we can test more easily as well.

**Comment by situchan** [comment]
@gedu please merge main

Build not posted 🤔  Please kick off adhoc build again

**Review by situchan — CHANGES_REQUESTED** [review]
There's no padding


https://github.com/Expensify/App/assets/108292595/8db974ee-ca4e-437a-8ed6-90673b45fc24

**Comment by grgia** [comment]
@situchan not sure what caused that bug- mine look alright

image

**Comment by grgia** [comment]
oof I pushed to my old branch, give me a sec!

**Comment by shawnborton** [comment]
Cool cool, and before we merge, can we do a test build so we can take these for a spin with our account data first? Thanks!

**Comment by grgia** [comment]
Updated and triggered a test build

**Comment by grgia** [comment]
Ready for review via test build @shawnborton @situchan

**Comment by shawnborton** [comment]
Testing now!

**Comment by shawnborton** [comment]
I see a tiny faint little corner outline in the bottom right near the receipt counter:
CleanShot 2024-02-14 at 15 12 57@2x

Otherwise this looks really nice!

**Comment by grgia** [comment]
I'm at a loss for what caused that tiny line in this PR, it seems to be due to the border/overflow but I didn't really update anything except the radius on the image component. Do you have any ideas @situchan ?

**Comment by situchan** [comment]
Let me check

**Comment by grgia** [comment]
Updating today

**Comment by shawnborton** [comment]
One thing I saw today was that when you hover over a row with a Split preview, the border outline on the split avatars unnecessarily changes. So maybe you can tackle that too?

**Comment by grgia** [comment]
Looking at that now @shawnborton !

**Comment by grgia** [comment]
Rebuilding another test build after the merge

**Comment by grgia** [comment]
image

All that's left is this weird bug- no clue what's causing that in this PR still

**Comment by grgia** [comment]
Looks like this is happening on DEV/main. Would you mind confirming @situchan and if so, we can move forward for now?
https://github.com/Expensify/App/assets/38015950/a1d999be-68b8-4ece-82a5-f3a73502b1c5

**Comment by situchan** [comment]
yes happening on main. Agree out of scope

old

**Comment by situchan** [comment]
Comparison:

### ReportPreview

Before:

old

After:

new

### ReportPreview (hover)

Before:

old-hover

After:

new-hover

### MoneyRequestPreview

Before:

old2

After:

new2

### MoneyRequestPreview (hover)

Before:

old2-hover

After:

new2-hover

File: src/components/ReportActionItem/ReportPreview.tsx
Comment (situchan) [inline src/components/ReportActionItem/ReportPreview.tsx:138]: Is it considered as pending receipt when `cardID` exists in `transaction`?
Also, should then name be `numberOfPendingRequests`? or this filters receipts only?

File: src/components/ReportActionItem/MoneyRequestPreview/MoneyRequestPreviewContent.tsx
Comment (situchan) [inline src/components/ReportActionItem/MoneyRequestPreview/MoneyRequestPreviewContent.tsx:258]: Is this intentional?

File: src/components/ReportActionItem/MoneyRequestPreview/MoneyRequestPreviewContent.tsx
Comment (situchan) [inline src/components/ReportActionItem/MoneyRequestPreview/MoneyRequestPreviewContent.tsx:223]: lineHeight was reduced by 4px. Hope this would not cut text in some devices 🙏

**Comment by shawnborton** [comment]
Lookin' good! cc @Expensify/design for any feedback too!

File: src/components/ReportActionItem/ReportPreview.tsx
Comment (grgia) [reply src/components/ReportActionItem/ReportPreview.tsx:138]: good catch, the variable name should be `numberOfPendingRequests`

File: src/components/ReportActionItem/MoneyRequestPreview/MoneyRequestPreviewContent.tsx
Comment (grgia) [reply src/components/ReportActionItem/MoneyRequestPreview/MoneyRequestPreviewContent.tsx:223]: Yes, this is the new proposed design. It should be fine for this text size 🤞

File: src/components/ReportActionItem/ReportPreview.tsx
Comment (situchan) [inline src/components/ReportActionItem/ReportPreview.tsx:155]: Hope there will be no case of this value being negative.
`reportPreviewAction.childMoneyRequestCount` - numberOfScanningReceipts - numberOfPendingRequests

**Comment by situchan** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>


https://github.com/Expensify/App/assets/108292595/f3942516-60da-42e2-92f8-01fe071a9dfe



</details>

<details>
<summary>Android: mWeb Chrome</summary>


https://github.com/Expensify/App/assets/108292595/cd6c2f15-6a98-4cbf-8933-4f38700904ab



</details>

<details>
<summary>iOS: Native</summary>


https://github.com/Expensify/App/assets/108292595/f720e713-1021-413c-a149-329786f02951



</details>

<details>
<summary>iOS: mWeb Safari</summary>


https://github.com/Expensify/App/assets/108292595/38a27419-0d46-470f-9b66-bc72bbb6d8dd



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>


https://github.com/Expensify/App/assets/108292595/08eb31f4-62c0-412b-a1cb-9bead4c6f6e7



</details>

<details>
<summary>MacOS: Desktop</summary>


https://github.com/Expensify/App/assets/108292595/3cb04780-1c35-49ba-81af-c9160a3d187a



</details>

**Review by situchan — APPROVED** [review]
@grgia please merge main to fix TS error

**Review by hayata-suenaga — DISMISSED** [review]
great work everyone! 🟢

**Comment by situchan** [comment]
@hayata-suenaga I see that @grgia is OOO this week. Which option do you suggest?
- Wait @grgia to pull main next week
- pull main yourself and merge
- merge as is since TS error is unrelated (fixed in https://github.com/Expensify/App/pull/37063)

**Comment by hayata-suenaga** [comment]
there is a merge freeze going on right now until next week anyway

so let's wait for @grgia to come back and merge main

**Comment by grgia** [comment]
Thanks everyone for your patience! I'll work on updating this PR

**Comment by grgia** [comment]
Going to rebuild the test build once tests pass

**Comment by situchan** [comment]
Can we create separate GH for this border issue? (if worth fixing)

border

**Review by situchan — APPROVED** [review]

**Comment by grgia** [comment]
Issue created: https://github.com/Expensify/App/issues/37485

**Comment by shawnborton** [comment]
Sorry for catching this so late but LAST piece of minor feedback: for the h1 we're using for the amount, can we use 28px for the line height instead of 32px?
CleanShot 2024-02-29 at 07 34 48@2x

Is 32 something specific to this component or is that a global value for h1 styles? If it's global, we can change it globally to 28px I think. cc @Expensify/design 

My thought is that bumping this down by 4px makes it appear like there's slightly less space below the amount, which is a good thing when we have a simple preview row.

You can see the difference here (bottom one has smaller line height):
CleanShot 2024-02-29 at 07 36 10@2x

**Comment by grgia** [comment]
@shawnborton I actually was about to message you to double check on that lineheight, I'll push that now!

**Comment by shawnborton** [comment]
H1 sounds good, but I think the font size should be 22px based on our Figma designs. So we might need to update that globally too.

**Comment by dannymcclain** [comment]
I'm good with those updates to the H1 even if they need to be made globally. @Expensify/design We should make sure our type styles in Figma have explicit line heights set so we all know what they should be (H1 & H2 styles currently have their line-heights set to `Auto`)

**Comment by shawnborton** [comment]
Agree! I think we should do 28px line height for h1, and 24px line height for h2. I can push those changes to Figma quickly!

**Review by hayata-suenaga — APPROVED** [review]
🟢 code looks okay
I'll wait for design team's final review of the latest build

**Review by bondydaa — APPROVED** [review]

**Comment by situchan** [comment]
Here's video of visual difference (main vs this branch)


https://github.com/Expensify/App/assets/108292595/d1cb619f-1e17-48c8-baeb-09510cab43be

**Comment by grgia** [comment]
Thank you everyone, going to merge this one!

**Comment by kavimuru** [comment]
@grgia could you help with pending expensify card transaction?

**Comment by kavimuru** [comment]
@roryabraham @situchan We are not sure how to have "pending expensify card transactions", It would be great if you could validate internally. Thanks!

**Comment by grgia** [comment]
**QA**
image
image
image

**Pull request merged**
